app/models/post.rb

```diff
@@ -2,4 +2,14 @@ class Post < ApplicationRecord
   belongs_to :user
   has_many :comments
   has_many :likes
+
+  after_save :posts_counter
+
+  def posts_counter
+    user.update(post_counter: user.posts.count)
+  end
```

Comment:
- [ ] [OPTIONAL] Well done for using the Rails `after_save` active callback.
A nice and more efficient way to implement the method that updates your post_counter would be:
```ruby
def posts_counter
  user.increment!(:post_counter)
end
```
You can apply this technique to the `likes_counter` and `comments_counter` as well